### PR TITLE
add contract address for token on account page

### PR DESCRIPTION
**Pull request by valpinkman**
Show the contract address on account page for token and link to explorer

### Type

Feat

### Context

LL-1578

### Parts of the app affected / Test plan

Account Detail Page

image

image

### CODE REVIEW

**Comment by valpinkman** [comment]
#update

image



image

File: src/components/AccountPage/AccountHeader.js [hunk 1]
```diff
@@ -29,26 +77,56 @@ const AccountName = styled(Text).attrs({
...
+    const explorerView = getDefaultExplorerView(mainAccount.currency)
+
+    const getContract = () =>
+      account.type === 'TokenAccount' && parentAccount
+        ? getAccountContractExplorer(explorerView, account, parentAccount)
+        : null
+
+    const contract = getContract()
+
+    const openLink = () => {
```
Comment (gre) [inline src/components/AccountPage/AccountHeader.js]: useCallback(..., [contract])

File: src/components/AccountPage/AccountHeader.js [hunk 2]
```diff
@@ -29,26 +77,56 @@ const AccountName = styled(Text).attrs({
...
     return (
       <Box horizontal align="center" flow={2} grow>
         <Box>
           <ParentCryptoCurrencyIcon currency={currency} borderColor="lightGrey" />
         </Box>
         <Box grow>
-          <CurName>{account.type === 'Account' ? currency.name : 'token'}</CurName>
+          {contract && account.type === 'TokenAccount' ? (
+            <Box horizontal alignItems="center">
+              <CurNameToken>{`${t('account.contractAddress')}`}</CurNameToken>
```
Comment (gre) [inline src/components/AccountPage/AccountHeader.js]: you could use `<Trans` to not have to pass in `t={t}`

File: yarn.lock [hunk 3]
```diff
@@ -14155,6 +14147,14 @@ redux@^4.0.1:
     loose-envify "^1.4.0"
     symbol-observable "^1.2.0"
 
+redux@^4.0.2:
```
Comment (gre) [inline yarn.lock]: might want to bump redux to 4.0.2 in desktop too to avoid dup

File: src/components/AccountPage/AccountHeader.js (hunk 1 above)
Comment (valpinkman) [reply src/components/AccountPage/AccountHeader.js]: done

File: src/components/AccountPage/AccountHeader.js (hunk 2 above)
Comment (valpinkman) [reply src/components/AccountPage/AccountHeader.js]: done

File: yarn.lock (hunk 3 above)
Comment (valpinkman) [reply yarn.lock]: bumped

**Comment by gre** [comment]
the bump of redux is still problematic because live-common and redux are sitll not on same in your fixes. i suggest we HODL for #2176 to gets in as it also have this fixed and with latest redux version.

**Pull request merged**